include/net/net_if.h

```diff
@@ -65,6 +65,9 @@ struct net_if_addr {
 	/** What is the current state of the address */
 	enum net_addr_state addr_state;
 
+	/** Reference counter */
```

Comment:
What is it fixing?

IP addresses are set either automatically or manually. Why would we remove it if there is no "users"?

Reply:
> IP addresses are set either automatically or manually. Why would we remove it if there is no "users"?

The removal only happens when application calls `net_if_addr_[add|rm]()`, nothing is done without user action. I hope I did not write somewhere that we would remove things without application initiating it, that needs to be fixed then.